frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx

```diff
@@ -0,0 +1,287 @@
+import { Monaco } from '@monaco-editor/react'
+import { IconPencil, IconPlus, IconX } from '@posthog/icons'
+import { LemonButton, LemonCheckbox, LemonInput, LemonSelect } from '@posthog/lemon-ui'
+import { useValues } from 'kea'
+import { CodeEditor } from 'lib/components/CodeEditors'
+import { languages } from 'monaco-editor'
+import { useEffect, useMemo, useState } from 'react'
+
+import { groupsModel } from '~/models/groupsModel'
+import { HogFunctionInputSchemaType } from '~/types'
+
+export type HogFunctionInputProps = {
+    schema: HogFunctionInputSchemaType
+    value?: any
+    onChange?: (value: any) => void
+    disabled?: boolean
+}
+
+const SECRET_FIELD_VALUE = '********'
+
+function useAutocompleteOptions(): languages.CompletionItem[] {
+    const { groupTypes } = useValues(groupsModel)
+
+    return useMemo(() => {
+        const options = [
+            ['event', 'The entire event payload as a JSON object'],
+            ['event.name', 'The name of the event e.g. $pageview'],
+            ['event.distinct_id', 'The distinct_id of the event'],
+            ['event.timestamp', 'The timestamp of the event'],
+            ['event.url', 'URL to the event in PostHog'],
+            ['event.properties', 'Properties of the event'],
+            ['event.properties.<key>', 'The individual property of the event'],
+            ['person', 'The entire person payload as a JSON object'],
+            ['project.uuid', 'The UUID of the Person in PostHog'],
+            ['person.url', 'URL to the person in PostHog'],
+            ['person.properties', 'Properties of the person'],
+            ['person.properties.<key>', 'The individual property of the person'],
+            ['project.id', 'ID of the project in PostHog'],
+            ['project.name', 'Name of the project'],
+            ['project.url', 'URL to the project in PostHog'],
+            ['source.name', 'Name of the source of this message'],
+            ['source.url', 'URL to the source of this message in PostHog'],
+        ]
+
+        groupTypes.forEach((groupType) => {
+            options.push([`groups.${groupType.group_type}`, `The entire group payload as a JSON object`])
+            options.push([`groups.${groupType.group_type}.name`, `Display name of the group`])
+            options.push([`groups.${groupType.group_type}.url`, `URL to the group in PostHog`])
+            options.push([`groups.${groupType.group_type}.properties`, `Properties of the group`])
+            options.push([`groups.${groupType.group_type}.properties.<key>`, `The individual property of the group`])
+            options.push([`groups.${groupType.group_type}.index`, `Index of the group`])
+        })
```

Comment:
Probably worth adding the name of the group somewhere in the descriptions, as I'd never know which one is `0` and which is `1`...

Reply:
The `group_type` is the key like `organization`. The `index` is the numeric ID

Reply:
Although this did make me notice there is a bug...